Use `pypyX.Y` for PyPy `python-version` input (#349)
actions/setup-python: README.md

- Preinstalled versions of PyPy in the tools cache on GitHub-hosted runners
  - For detailed information regarding the available versions of PyPy that are installed, see [Supported software](https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#supported-software).
  - For the latest PyPy release, all versions of Python are cached.

>   - Cache is updated with a 1-2 week delay. If you specify the PyPy version as `pypy3.7`, the cached version will be used although a newer version is available. If you need to start using the recently released version right after release, you should specify the exact PyPy version using `pypy3.7-v7.3.3`.

Reviewer: ```suggestion
  - Cache is updated with a 1-2 week delay. If you specify the PyPy version as `pypy3.7` or `pypy-3.7`, the cached version will be used although a newer version is available. If you need to start using the recently released version right after release, you should specify the exact PyPy version using `pypy3.7-v7.3.3` or `pypy-3.7-v7.3.3`.
```

  - Using the most recent patch version will result in a very quick setup since no downloads will be required since a locally installed version Python on the runner will be used.

# Specifying a PyPy version
The version of PyPy should be specified in the format `pypy<python_version>[-v<pypy_version>]` or `pypy-<python_version>[-v<pypy_version>]`.
The `<pypy_version>` parameter is optional and can be skipped. The latest version will be used in this case.

```

> pypy3.7 # the latest available version of PyPy that supports Python 3.7
> pypy3.8 # the latest available version of PyPy that supports Python 3.8
> pypy2.7 # the latest available version of PyPy that supports Python 2.7
> pypy3.7-v7.3.3 # Python 3.7 and PyPy 7.3.3
> pypy3.7-v7.x # Python 3.7 and the latest available PyPy 7.x
> pypy3.7-v7.3.3rc1 # Python 3.7 and preview version of PyPy
> pypy3.7-nightly # Python 3.7 and nightly PyPy

Reviewer: ```suggestion
pypy-3.7 or  pypy3.7 # the latest available version of PyPy that supports Python 3.7
pypy-3.8 or pypy3.8 # the latest available version of PyPy that supports Python 3.8
pypy-2.7 or pypy2.7 # the latest available version of PyPy that supports Python 2.7
pypy-3.7-v7.3.3 or pypy3.7-v7.3.3 # Python 3.7 and PyPy 7.3.3
pypy-3.7-v7.x or pypy-3.7-v7.x # Python 3.7 and the latest available PyPy 7.x
pypy-3.7-v7.3.3rc1 or pypy-3.7-v7.3.3rc1 # Python 3.7 and preview version of PyPy
pypy-3.7-nightly or pypy-3.7-nightly # Python 3.7 and nightly PyPy
```

  - Using the most recent patch version will result in a very quick setup since no downloads will be required since a locally installed version Python on the runner will be used.

# Specifying a PyPy version

> The version of PyPy should be specified in the format `pypy<python_version>[-v<pypy_version>]` or `pypy-<python_version>[-v<pypy_version>]`.

Author: @dmitry-shibanov,

I think that the proposed changed throughout the documentation mentioning both specifiers each time might just confuse the reader.
IMHO, only one kind of specifier should appear in the docs except in one place.
As mentioned in #346 rationale, most of the ecosystem uses `pypyX.Y` and that's what I went with.

Maybe the following line might even be a bit more harsh on `pypy-X.Y` future ?

```suggestion
The version of PyPy should be specified in the format `pypy<python_version>[-v<pypy_version>]`. The format  `pypy-<python_version>[-v<pypy_version>]` is deprecated (still supported for backward compatibility but might be dropped in a future version).
```

Reviewer: Hello @mayeut. For now we do not have plans to deprecate it.

Author: @dmitry-shibanov, no problem. But, if you have to recommend one (and that's what the doc should be doing IMHO), which one ? Otherwise it just clobbers the doc and the reader doesn't know why there are 2 specifiers mentioned everywhere ?

Reviewer: For now I think it's better to keep both variants in the documentation. I think that removing `pypy-` can lead to misconception.

Author: ok will do
